synapse/rest/media/v1/_base.py

```diff
@@ -96,7 +113,15 @@ def add_file_headers(request, media_type, file_size, upload_name):
     def _quote(x):
         return urllib.parse.quote(x.encode("utf-8"))
 
-    request.setHeader(b"Content-Type", media_type.encode("UTF-8"))
+    # will only fire for unspecified charsets, i.e.,
```

Comment:
```suggestion
    # Default to a UTF-8 charset for text content types.
    # 
    # This will only fire if no charset (or any other parameter) is given: ie, it will match
```